onnxruntime/core/session/inference_session.cc

```diff
@@ -528,6 +528,11 @@ common::Status InferenceSession::Initialize() {
     // now that all the transforms are done, call Resolve on the main graph. this will recurse into the subgraphs.
     ORT_RETURN_IF_ERROR(graph.Resolve());
 
+    if (!session_options_.optimized_model_filepath.empty()) {
+      // Serialize optimized onnx model.
+      Model::Save(*model_, session_options_.optimized_model_filepath);
```

Comment:
Return value from Save call should be checked.

Reply:
Ok, will fail if model save failed.